pkg/api/v1/types.go

```diff
@@ -1706,6 +1706,20 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty"`
 }
 
+// DaemonEndpoint stores information about a single Daemon endpoint.
+type DaemonEndpoint struct {
+	// Port number of the given endpoint.
+	Port int `json:port`
+}
+
+// NodeDaemonEndpoints lists ports opened by daemons running on the Node.
+type NodeDaemonEndpoints struct {
+	// Endpoint on which Kubelet is listening.
+	KubeletEndpoint DaemonEndpoint `json:"kubeletEndpoint,omitempty"`
+	// Endpoint opened by Kubelet on which cAdvisor proxy is available.
+	CAdvisorProxyEndpoint DaemonEndpoint `json:"cAdvisorProxyEndpoint,omitempty"`
```

Comment:
@gmarek - Similar to the read-only kubelet port, I'm going to push back on adding the cadvisor proxy port into the v1 API. 

Every port that we expose on the kubelet makes the work on securing the kubelet significantly more difficult. We need to configure authn/z handlers, set up SSL and potentially configure firewall rules to allow access. 

What consumers of the NodeDaemonEndpoints struct are going to use the cadvisor port? Do you know which system components hit cadvisor directly today?

/cc @vishh